### PR TITLE
adding MinimalApi and Area commands

### PR DESCRIPTION
**Pull request by deepchoudhery**
Adding `MinimalApiCommand` and `AreaCommand` (help at the bottom)
- MinimalApi has 2 different templates, one supports a DbContext (EF scenario), and one without.
  - `MinimalApi.tt` and `MinimalApiEf.tt`  templates
- scenarios covered and tested:
  - empty .NET apps (Web App, Web API) 
  - .NET apps with Individual Auth (this app has EF package references and a DbContext already)
  - with and without DbContexts
  - new and existing DbContexts
- missing scenarios :
  - package installation is busted rn due to incorrectly registering MSBuild (fix upcoming soon)
  - TypedResult support
  - [PrimaryKey] attributed keys and composite primary keys (following the official EF Core [docs ](https://learn.microsoft.com/en-us/ef/core/modeling/keys?tabs=data-annotations)for primary key discovery) are not yet detected.

minor changes: 
- updated `Microsoft.Build` and `Microsoft.Build.Locator` package versions
- remove `InteractivePickerType.DbProviderPicker` and providing DbProvider selection
- rename `EfDbContextAnalyzer` --> `EfDbContextHelpers`
- added DbSet statement to `NewDbContext.tt`

### CODE REVIEW

File: src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Helpers/Roslyn/ProjectModifierHelper.cs [hunk 1]
```diff
@@ -466,6 +466,11 @@ internal static bool FilterOptions(string[]? options, CodeChangeOptions codeChan
         {
             return false;
         }
+        //if the CodeChange in the config is for an 'EfScenario' and
```
Comment (tlmii) [inline src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Helpers/Roslyn/ProjectModifierHelper.cs:469]: incomplete comment?

File: src/dotnet-scaffolding/dotnet-scaffold-aspnet/dotnet-scaffold-aspnet.csproj [hunk 2]
```diff
@@ -23,4 +29,36 @@
...
+    </Compile>
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Pack="true" Include="$(RepoRoot)src\dotnet-scaffolding\dotnet-scaffold-aspnet\Templates\*\*.tt" PackagePath="Templates\" />
+    <None Pack="true" Include="$(RepoRoot)src\dotnet-scaffolding\Microsoft.DotNet.Scaffolding.Helpers\Templates\DbContext\*.tt" PackagePath="Templates\DbContext\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="C:\Users\decho\scaffolding-PR-2\scaffolding\src\dotnet-scaffolding\dotnet-scaffold-aspnet\Templates\MinimalApi\MinimalApiEf.tt">
```
Comment (tlmii) [inline src/dotnet-scaffolding/dotnet-scaffold-aspnet/dotnet-scaffold-aspnet.csproj:58]: Local path snuck in here

Comment (deepchoudhery) [reply src/dotnet-scaffolding/dotnet-scaffold-aspnet/dotnet-scaffold-aspnet.csproj:58]: oops fixed tyty

**Review by tlmii — APPROVED** [review]
I didn't focus too much on the actual generated content (the .tt files), assuming that's mostly ported. The rest seems to make sense and fit in with what I'd expect. A couple small questions. Approving so you can move forward with the fix for the package install.

File: src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Helpers/General/DotnetCliRunner.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ internal class DotnetCliRunner
 {
     public static DotnetCliRunner CreateDotNet(string commandName, IEnumerable<string> args)
     {
-        return Create(DotNetMuxer.MuxerPathOrDefault(), new[] { commandName }.Concat(args));
+        return Create("dotnet", new[] { commandName }.Concat(args));
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Helpers/General/DotnetCliRunner.cs:14]: yep, so far I've kept it as simple as creating a ProcessInfo with `FileName` as the command name ('dotnet' in this case) but we can make this more robust for sure.

File: eng/Versions.props [hunk 4]
```diff
@@ -70,10 +70,10 @@
     <!-- Microsoft.AspNetCore.Razor.Language -->
     <MicrosoftAspNetCoreRazorLanguagePackageVersion>6.0.24</MicrosoftAspNetCoreRazorLanguagePackageVersion>
     <!-- Microsoft.Build-->
-    <MicrosoftBuildPackageVersion>17.9.5 </MicrosoftBuildPackageVersion>
+    <MicrosoftBuildPackageVersion>17.10.4 </MicrosoftBuildPackageVersion>
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply eng/Versions.props:73]: by these 2 not sure if you mean CodeAnalysis and Microsoft.Build or Microsoft.Build and Microsoft.Build.Locator? regardless I'll double check in either cases if there are dependencies or need to be versioned together first (always just update them when I can)

**Pull request merged**